src/main/java/com/jmatio/types/MLStructureObjectBase.java

```diff
@@ -81,7 +80,7 @@ public void setField(String name, MLArray value, int index) {
 
 		Map<String, MLArray> map = mlStructArray.get(index);
 		if (map == null) {
-			map = new LinkedHashMap<String, MLArray>();
+			map = new TreeMap<String, MLArray>();
 			mlStructArray.put(index, map);
```

Comment:
`TreeMap` is an excellent choice, but rather than using the natural order of String, it should use the explicit ordering laid out in `keys`.  Guava's [ExplicitOrdering](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/ExplicitOrdering.java) is an example implementation.